components/script/dom/formdata.rs

```diff
@@ -55,24 +48,34 @@ impl FormData {
 
 impl FormDataMethods for FormData {
     // https://xhr.spec.whatwg.org/#dom-formdata-append
-    fn Append(&self, name: USVString, value: USVString) {
+    fn Append(&self, name: USVString, str_value: USVString) {
+        let datum = FormDatum {
+            ty: DOMString::from(""), // XXX(izgzhen): ?
```

Comment:
Nope .. My bad, spec says:

> For the purposes of interaction with other algorithms, an entry's type is "string" if its value is a string and "file" otherwise. If an entry's type is "file", its filename is the empty string if value is not a File object, and otherwise its filename is the value of entry's value's name attribute.

https://xhr.spec.whatwg.org/#formdata